### ISSUE TITLE
vector tiles in mbtiles format not exported

### ISSUE THREAD

**Issue by ffinger** [opengisch/qfieldsync#381]
Hi all,

I am working on setting up a project including vector tiles in the mbtiles format from here: https://data.maptiler.com/downloads/dataset/osm/africa/congo-democratic-republic
The tile layer includes OSM vector data for a large region and is quite heavy (>300 Mb). Since I will be working offline I downloaded the mbtiles file and included it in my QGis project, which works well. I made sure the file is included (`no action`) in the QFieldSync project properties.

Once opening the export on my Android device however I get a message saying that my mbtile layer is missing:

>The following layers could not be loaded, please review those and reconfigure the QGis project.
>`type=mbtiles&url=**PATH TO FILE ON MY PC**`

The file is indeed not in the exported folder. When I check the QGis project (.qgs) file created it still contains the original path to the mbtile file on my computer. Adding the file to the exported folder manually and changing the path makes work everything perfectly well and smoothly on my Android device.

Is there a reason why mbtiles files are not included in the export (e.g. to prevent users from exporting very heavy files)? Or is this a bug?

Thanks!

PS: just saying, QField is brilliant!

**Comment by nirvn** [comment]
@ffinger , are you running the latest versions of qfieldsync and QGIS? We fixed this issue a while back. A new regression might have slipped in but first like make sure you try using latest builds.

Are you doing cloud export or USB cable export? If the latter, the right action would be to copy.

**Comment by ffinger** [comment]
Hi @nirvn.
I am using QGIS 3.16.16 LTS & QFieldSync 3.4.4
I could probably update QGIS to 3.2 but would prefer to stick to the LTS version. Do you think this would make a difference?

**Comment by ffinger** [comment]
Just for completeness, I installed the beta version of QFieldSync. The issue is still there, although now I can chose `copy`. The layer is copied but the path in the exported qgis project is wrong.

**Comment by nirvn** [comment]
I'd try with 3.22 if I were you. I don't think the fix was backported to 3.16 (which will be replaced by 3.22 as LTS within 2 weeks btw :) )

**Comment by m-kuhn** [comment]
@SeqLaz can you try to reproduce this locally?

**Comment by SeqLaz** [comment]
> @SeqLaz can you try to reproduce this locally?

@m-kuhn @InsertUser In the test I conducted, I used the MBTile with the Copy option in the QField tab of the Project properties. I packaged the project, and the MBTiles use relative paths. You can see the steps I followed in the attached document.

Could you check if you have specified the Project Home in the Project properties under the General section?



[Test vector tiles in mbtiles format not exported 001.pdf](https://github.com/user-attachments/files/17850967/Test.vector.tiles.in.mbtiles.format.not.exported.001.pdf)

**Comment by InsertUser** [comment]
OK some weirdness going on here. I forgot to mention that I'm on Windows 11 with version 4.11.0 of the plugin (installed today).

I am using the "experimental" mvt tiles from [here](https://download.geofabrik.de/europe/united-kingdom/wales.html) for testing. Saved to the same folder as the test file. They look OK in QGIS (other than missing ocean at some zoom levels). 

My file properties are set to relative file paths and for Cable Export the layer is to be copied. 

If I export _without_ styling the tiles then if I look in the resulting qgs file with a text editor I see two instances of the string ".mbtiles":
``` 
	Line  38:     <layer-tree-layer patch_size="-1,-1" expanded="1" checked="Qt::Checked" providerKey="mbtilesvectortiles" legend_exp="" source="type=mbtiles&amp;url=./wales-shortbread-1.0.mbtiles" id="wales_shortbread_1_0_da721106_6ec4_46f5_8cf6_6cad0714b8eb" legend_split_behavior="0" name="wales-shortbread-1.0">
	Line 161:       <datasource>type=mbtiles&amp;url=./wales-shortbread-1.0.mbtiles</datasource>
```

This works OK if copied to my Android tablet. 

If I apply the [Versatiles](https://github.com/versatiles-org/versatiles-style) greybeard style (which has some visible translation issues but no error message/warning) and then export the resulting file has two different paths in the two locations:
```
	Line    38:     <layer-tree-layer source="type=mbtiles&amp;url=./wales-shortbread-1.0.mbtiles" expanded="1" patch_size="-1,-1" providerKey="mbtilesvectortiles" id="wales_shortbread_1_0_da721106_6ec4_46f5_8cf6_6cad0714b8eb" name="wales-shortbread-1.0" legend_exp="" legend_split_behavior="0" checked="Qt::Checked">
	Line   161:       <datasource>type=mbtiles&amp;url=../../../OneDrive/osm/test/qfield%20wales%20mbtiles%20mvt/wales-shortbread-1.0.mbtiles</datasource>
```

When copied to my tablet this second version (with the mismatched paths) gives this error:
> Unable to load some layers
> The following layers could not be loaded, please review those and reconfigure the QGIS project. 
> wales-shortbread-1.0
> `type=mbtiles&url=../../../OneDrive/osm/test/qfield%20wales%20mbtiles%20mvt/wales-shortbread-1.0.mbtiles`

If I copy that directory and open the file on my desktop I can inspect the layer and see that the source in indeed set to the old location. After manually correcting it to the correct location in the copied folder and saving the project file it then works fine when copied to my device. 

Why would a style cause the location to fail to be updated when the project gets repackaged?

**Comment by m-kuhn** [comment]
Could you share your (unexported) project?

**Comment by InsertUser** [comment]
Today is a day where I felt like I'm going crazy. 

As there doesn't appear to be a way to save a file without leaking personal information that makes me uncomfortable I tried to recreate the problem file on a system that doesn't have any. I was unable to create a new file with this problem either on the second system or the original Windows system no matter what I tried. It seemed so easy to recreate the error yesterday. 

I think the (unexported) files I created yesterday must have been corrupted somehow and that's why yesterday's files are the only ones I can export broken today. Everything new exports fine even though I think I made them in the exact same way. 

I can only guess that my computer somehow got in a weird state yesterday and was somehow misbehaving when it saved project files. I am embarrassed to think that it may have been me shutting my computer down overnight that fixed the issue.
